src/storage/storage.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+    Injectable,
+    InternalServerErrorException,
+    Logger,
+    LoggerService,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+    ReadStream,
+    WriteStream,
+    createReadStream,
+    createWriteStream,
+} from 'fs';
+import { readdir } from 'fs/promises';
+import { join } from 'path';
+import { v4 as uuid } from 'uuid';
+import * as fs from 'fs';
+
+@Injectable()
+export class StorageService {
+    constructor(private readonly configService: ConfigService) {}
+
+    async saveFile(file: Express.Multer.File): Promise<string> {
+        const storage: string | undefined =
```

Comment:
Zastanawiam się czy przyjmujemy plik Multerem czy stream i robimy to przez pipe, skoro i tak tworzysz writeStream? Fakt, że to będą małe pliki więc mało co zamulamy ale skoro i taj jest writeStream to można przyjmować streami i od razu po kawałku go zapisywać ? Sebastian?

Reply:
W sumie to nie wiem o co chodzi w tym komentarzu. Wybrałem Multer.File ponieważ możemy wykorzystać wtedy FileInterceptor w kontrolerze, gdzie zrobimy walidacje pliku. Dodatkowo Multer wymaga enctype="multipart/form-data". Więc kolejne zabezpieczenie. OD razu możemy zapisać ten plik w tym kontrolerze, albo wysłać go do service.

Reply:
no to właśnie o to mi chodziło :) 
walidacja na poziomie kontrolera, kiedy będziemy dostawć plik